### PR TITLE
types gpt: hash long device names

### PR DESCRIPTION
**Pull request by Lassulus**
this should mitigate some of the problems in https://github.com/nix-community/disko/issues/551

also we fix the test framework with recent versions of nixpkgs

### CODE REVIEW

File: lib/types/gpt.nix [hunk 1]
```diff
@@ -55,7 +55,13 @@ in
           };
           label = lib.mkOption {
             type = lib.types.str;
-            default = "${config._parent.type}-${config._parent.name}-${partition.config.name}";
+            default = let
+              limit = 36; # not sure why it's the limit but it seems to break with 37 characters
```
Comment (Mic92) [inline lib/types/gpt.nix:59]: For GPT it's actually 72 characters inside the label?

Comment (Mic92) [reply lib/types/gpt.nix:59]: Ah, found it: https://en.wikipedia.org/wiki/GUID_Partition_Table


56 (0x38) 	72 bytes 	Partition name (36 [UTF-16](https://en.wikipedia.org/wiki/UTF-16)LE code units)

Comment (Mic92) [reply lib/types/gpt.nix:59]: So fdisk encodes the name as UTF-16.

File: lib/tests.nix [hunk 2]
```diff
@@ -223,10 +223,15 @@ let
                   disk_flags += f' -drive file={oldmachine.state_dir}/empty{i}.qcow2,id=drive{i + 1},if=none,index={i + 1},werror=report'
                   disk_flags += f' -device virtio-blk-pci,drive=drive{i + 1}'
               return disk_flags
-          def create_test_machine(oldmachine=None, args={}): # taken from <nixpkgs/nixos/tests/installer.nix>
+          def create_test_machine(oldmachine, args={}): # taken from <nixpkgs/nixos/tests/installer.nix>
+              startCommand = "${pkgs.qemu_test}/bin/qemu-kvm"
+              startCommand += " -cpu max -m 1024 -virtfs local,path=/nix/store,security_model=none,mount_tag=nix-store"
+              startCommand += disks(oldmachine, ${toString num-disks})
+              ${lib.optionalString efi ''
```
Comment (Mic92) [inline lib/tests.nix:230]: is the efi branched used in the test?

Comment (Lassulus) [reply lib/tests.nix:230]: Not sure if the new ones do it though

**Pull request merged**